test/libsolidity/syntaxTests/largeTypes/large_storage_structs.sol

```diff
@@ -0,0 +1,65 @@
+contract C {
+    struct P { uint256[2**63] x; }
+
+    struct S0 {
+        P[2**62] x;
+        P y;
+    }
+    S0 s0;
+
+    struct S1 {
+        P x;
+        P[2**62] y;
+    }
+    S1 s1;
+
+    struct S2 {
+        mapping(uint => P[2**62]) x;
+        mapping(uint => P[2**62]) y;
+        mapping(uint => S2) z;
+    }
+    S2 s2;
+
+    struct Q0
+    {
+        uint[1][][2**65] x;
+        uint[2**65][][1] y;
+        uint[][2**65] z;
+        uint[2**65][] t;
+    }
+    Q0 q0;
+
+    struct Q1
+    {
+        uint[1][][2**65] x;
+    }
+    Q1 q1;
+
+    struct Q2
+    {
+        uint[2**65][][1] y;
+    }
+    Q2 q2;
+
+    struct Q3
+    {
+        uint[][2**65] z;
+    }
+    Q3 q3;
+
+    struct Q4
+    {
+        uint[2**65][] t;
+    }
+    Q4 q4;
+}
+// ----
+// Warning 3408: (108-113): Variable covers a large part of storage and thus makes collisions likely. Either use mappings or dynamic arrays and allow their size to be increased only in small quantities per transaction.
```

Comment:
We cannot really see which lines these warnings refer to, i.e. which types are considered invalid and which are not. Can you split it up into multiple tests?

Reply:
I tried, but found that it might be better to have all messages together. It makes it easier to see how different warnings are produced.

So instead of splitting the test, I added a variable name to the warning. Also, I used types which are recognizable in the messages.

It would be no problem to add file splitting on top of it, but IMO it would be less convenient.

See PR #9311.